### PR TITLE
Feat home zuri

### PR DESCRIPTION
**Pull request by shalomtaiwo**
Add product category and filter

image

### CODE REVIEW

**Comment by shalomtaiwo** [comment]
https://jam.dev/c/994bb10a-4827-4a83-b7a4-1e06a4a68bb4

File: modules/home/products/SearchAndFilterProducts.tsx [hunk 1]
```diff
@@ -32,55 +43,40 @@ const SearchAndFilterProducts = (prop: {
...
-    if (option === 'clear') {
-      prop.setFilter({});
-    } else {
-      setShowFilterComponent(false);
-    }
-  };
   const slideLeft = () => {
-    const slider = sliderRef.current!; // Non-null assertion
-    slider.scrollLeft -= 150; // Adjust the scroll distance as needed
+    const slider = sliderRef.current!;
```
Comment (ickynavigator) [inline modules/home/products/SearchAndFilterProducts.tsx]: confirm that slider ref is not null instead

File: modules/home/products/SearchAndFilterProducts.tsx [hunk 2]
```diff
@@ -32,55 +43,40 @@ const SearchAndFilterProducts = (prop: {
...
+    slider.scrollLeft -= 150;
   };
 
   const slideRight = () => {
-    const slider = sliderRef.current!; // Non-null assertion
-    slider.scrollLeft += 150; // Adjust the scroll distance as needed
+    setShowLeftButton(true);
+    const slider = sliderRef.current!;
```
Comment (ickynavigator) [inline modules/home/products/SearchAndFilterProducts.tsx]: confirm that slider ref is not null instead

File: modules/home/products/DynamicCategoryCarousel.tsx [hunk 3]
```diff
@@ -235,6 +233,31 @@ const DynamicCategoryCarousel = () => {
                 )}
               </div>
             ))}
+
+            {isCategoryNamesLoading ||
+              (slides.length === 0 &&
+                sliders.map((category, index) => (
+                  <div key={index} className="relative h-[250px] sm:h-[300px] w-[182.71]">
```
Comment (ickynavigator) [inline modules/home/products/DynamicCategoryCarousel.tsx:238]: index is not a valid key

File: modules/home/products/SearchAndFilterProducts.tsx [hunk 4]
```diff
@@ -197,56 +337,115 @@ const SearchAndFilterProducts = (prop: {
...
+                  shop: any;
+                  rating: number | undefined;
+                  price: string;
+                  images: Array<{ url: string }>;
+                  name: string;
+                },
+                index: number,
+              ) => (
+                <SectionProductCard
+                  key={index}
```
Comment (ickynavigator) [inline modules/home/products/SearchAndFilterProducts.tsx]: don't use index as the key

File: modules/home/products/sectionData.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  setActiveSection,
+  handleFilters,
+  setShowFilterComponent,
+  fetchProducts,
+  fetchAllData,
+}) => {
+  return (
+    <div className="justify-start items-center inline-flex mt-4 gap-6">
+      <div
+        key={0}
```
Comment (ickynavigator) [inline modules/home/products/sectionData.tsx]: remove this key

File: modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx [hunk 6]
```diff
@@ -122,12 +124,43 @@ const DynamicPortfolioCarousel = () => {
...
+                  | Iterable<ReactNode>
+                  | ReactPortal
+                  | PromiseLikeOfReactNode
+                  | null
+                  | undefined;
+                skills: string[];
+              },
+              index: Key | null | undefined,
+            ) => (
               <div key={index} className="relative h-[250px] sm:h-[300px] w-[182.71]">
```
Comment (ickynavigator) [inline modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx]: don't use index as key

File: modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx [hunk 7]
```diff
@@ -122,12 +124,43 @@ const DynamicPortfolioCarousel = () => {
...
-          <Slider {...settings}>
-            {slides.map((logo, index) => (
+      <div className="overflow-hidden p-2 w-full mx-0 mt-[0]">
+        <Slider {...settings}>
+          {slides?.map(
+            (
+              logo: {
+                section: string;
+                src: any;
+                name:
```
Comment (ickynavigator) [inline modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx]: add the type to the dat object instead

the type union you are using is unnecessary

string|undefined is unnecessary

File: modules/home/products/CustomFilterDropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+          <div className="w-5 h-5 relative">
+            <ArrowDown2 color="#737373" size={16} />
+          </div>
+        </div>
+      </div>
+      {isOpen && (
+        <div className="list mt-2 bg-white shadow-lg hover:bg-white-100 max-h-fit overflow-y-auto absolute z-20 w-full bg-white-100 rounded-xl">
+          {options.map((option, index) => (
+            <div
+              key={index}
```
Comment (ickynavigator) [inline modules/home/products/CustomFilterDropdown.tsx]: don't use index as key

**Review by ickynavigator — CHANGES_REQUESTED** [review]

File: modules/home/products/CustomFilterDropdown.tsx (hunk 8 above)
Comment (shalomtaiwo) [reply modules/home/products/CustomFilterDropdown.tsx]: fixed

File: modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx (hunk 7 above)
Comment (shalomtaiwo) [reply modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx]: fixed

File: modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx (hunk 6 above)
Comment (shalomtaiwo) [reply modules/home/carousel/portfolios/DynamicPortfolioCarousel.tsx]: fixed

**Review by Samadeen — APPROVED** [review]

File: modules/home/products/sectionData.tsx (hunk 5 above)
Comment (shalomtaiwo) [reply modules/home/products/sectionData.tsx]: removed

**Review by Samadeen — APPROVED** [review]

File: modules/home/products/SearchAndFilterProducts.tsx (hunk 4 above)
Comment (shalomtaiwo) [reply modules/home/products/SearchAndFilterProducts.tsx]: fixed

File: modules/home/products/DynamicCategoryCarousel.tsx (hunk 3 above)
Comment (shalomtaiwo) [reply modules/home/products/DynamicCategoryCarousel.tsx:238]: used the category id instead

File: modules/home/products/SearchAndFilterProducts.tsx (hunk 2 above)
Comment (shalomtaiwo) [reply modules/home/products/SearchAndFilterProducts.tsx]: fixed

File: modules/home/products/SearchAndFilterProducts.tsx (hunk 1 above)
Comment (shalomtaiwo) [reply modules/home/products/SearchAndFilterProducts.tsx]: fixed

**Pull request merged**